AirSkyBoat/AirSkyBoat
# Adds furniture quests

**_I affirm:_**


## Please enter a player-facing description

Added Furniture Quests for the following items: Armoire, Beverage Barrel, Bureau, Copy of Lines and Spaces, Cupboard, Oak Bed, Simple Bed, Stationary Set, Tarutaru Desk, Water Cask, White Jar, Wicker Box.  (Tiberon)
Note: The quest Unexpected Treasure is not added in this PR.

## What does this pull request do? (Please be technical)

Adds a light framework via globals/furniture_quests.lua which implements retail era behavior/wait times for all known/captured furniture quests.
Adds an era module which updates wait times to be era appropropriate for LSB.

## Steps to test these changes

Without access to the database this becomes a bit of a pain.
it is possible to use `!setplayervar <player> <variable> <value>` to quicken testing.

Variables for furniture are as follows:
- [furnitureName]received - 1 if the non-repeatable reward has been gained. 0 otherwise.
- [furnitureName]rewardTime - the vanadiel epoc when the reward can be claimed
- [furnitureName]mustZone - a local var, set to signify the player must zone

Testing:
1. Place a furniture from the list above in your nations mog house
2. Zone out of mog house
3. Zone back into mog house
4. Talk to moogle when the vanadielEpoc is post rewardTime.  The easiest way to make this happen is to set the rewardTime player var to something very low like ~2
5. Collect reward
6. Verify cant get another reward on the char
7. Verify rewards
8. Verify repeatable rewards each conquest
9. Verify if inventory is full, the player will get the reward next chat

## Special Deployment Considerations

None
